docs/advanced/kotlin-codegen.md

```diff
@@ -0,0 +1,85 @@
+# About
+
+In addition to the Java code-gen, we also have an experimental Kotlin API that generates more idiomatic Kotlin classes. Instead of generating Java code that can be used in Kotlin, it generates native Kotlin code that leans into techniques that only Kotlin supports. This is split primarily into two parts: the data classes used to serialize responses (both in the server and in the client), and the client query projections
```

Comment:
Would it help to also clarify that you can generate kotlin classes using the regular approach (linking to it). Currently you mention Java codegen, btut would be helpful to clarify that it also generates kotlin. What you describe here is a more idiomatic apprach etc.

Reply:
The docs look great, thanks for adding!

Reply:
Updated